staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -504,7 +504,9 @@ func (r *Request) tryThrottleWithInfo(ctx context.Context, retryInfo string) err
 	now := time.Now()
 
 	err := r.rateLimiter.Wait(ctx)
-
+	if err != nil {
+		err = fmt.Errorf("client rate limiter Wait returned an error: %w", err)
```

Comment:
from down in rate.go

```
	// Check if ctx is already cancelled
	select {
	case <-ctx.Done():
		return ctx.Err()
	default:
	}
```

if we timed out and our last API request failed, it's more useful to return the last failure error.

Reply:
@deads2k https://github.com/kubernetes/kubernetes/pull/108698